nvtabular/ops/categorify.py

```diff
@@ -325,6 +325,8 @@ def fit(self, col_selector: ColumnSelector, ddf: dd.DataFrame):
                 for col in group.names:
                     self.storage_name[col] = name
 
+        ddf = dataset.to_ddf()
```

Comment:
This is the part I'm a little bit concerned about, because I'm worried that it will read more columns than are needed for the fitting. In order to make `fit` accept a `Dataset` I had to move the `to_ddf()` call into the operators, but we don't have all the information we'd need in order to determine which columns are used and which aren't.

The `ColumnSelector` is a start, but doesn't include additional dependency columns that are used but not transformed by the operator. That part lives in the `Node` input schema, which was accessible from the Workflow fitting code, but isn't from inside the operator.

So, this works, but there may be a performance regression here.